tests/model/test_model.py

```diff
@@ -2138,6 +2139,69 @@ def test_generate_all_emoji_data(self, mocker, model, zulip_emoji,
         # zulip_extra_emoji replaces all other emoji types for 'zulip' emoji.
         assert all_emoji_data['zulip']['type'] == 'zulip_extra_emoji'
 
+    @pytest.mark.parametrize('to_vary_in_realm_emoji', [
+            {
+                "deactivated": False,
+                "id": "20",
+                "name": "joy_cat",
+            },
+            {
+                "deactivated": True,
+                "id": "202020",
+                "name": "joker",
+            },
+            {
+                "deactivated": False,
+                "id": "22",
+                "name": "zulip",
+            },
+            {
+                "deactivated": True,
+                "id": "4",
+                "name": "zulip",
+            },
+            {
+                "deactivated": False,
+                "id": "23",
+                "name": "new_emoji",
+            },
+            {
+                "deactivated": True,
+                "id": "3",
+                "name": "singing",
+            },
+        ],
+        ids=[
+            'realm_emoji_with_same_name_as_unicode_emoji_added',
+            'realm_emoji_with_same_name_as_unicode_emoji_removed',
+            'realm_emoji_with_name_as_zulip_added',
+            'realm_emoji_with_name_as_zulip_removed',
+            'realm_emoji_added',
+            'realm_emoji_removed',
+        ]
+    )
+    def test__handle_update_emoji_event(self, mocker, model, zulip_emoji,
+                                        unicode_emojis, realm_emojis,
+                                        to_vary_in_realm_emoji):
+        realm_emojis[to_vary_in_realm_emoji['id']] = to_vary_in_realm_emoji
+        response = {
+            'type': 'realm_emoji',
+            'realm_emoji': realm_emojis
+        }
```

Comment:
This was broadly what I meant by a template. I might consider naming this `event` or similar; there's no request/response pair exactly, and we're testing an event-handler.